core/src/banking_stage/consumer.rs

```diff
@@ -392,14 +392,54 @@ impl Consumer {
         bank: &Arc<Bank>,
         txs: &[SanitizedTransaction],
         chunk_offset: usize,
+    ) -> ProcessTransactionBatchOutput {
+        // No filtering before QoS - transactions should have been sanitized immediately prior to this call
+        let pre_results = std::iter::repeat(Ok(()));
+        self.process_and_record_transactions_with_pre_results(bank, txs, chunk_offset, pre_results)
+    }
+
+    pub fn process_and_record_aged_transactions(
+        &self,
+        bank: &Arc<Bank>,
+        txs: &[SanitizedTransaction],
+        max_slot_ages: &mut [Slot],
+    ) -> ProcessTransactionBatchOutput {
+        // Need to filter out transactions since they were sanitized earlier.
+        // This means that the transaction may cross and epoch boundary (not allowed),
+        //  or account lookup tables may have been closed.
+        let pre_results = txs
+            .iter()
+            .zip(max_slot_ages)
+            .map(|(tx, max_slot_age)| {
+                if *max_slot_age < bank.slot() {
+                    *max_slot_age = 0; // Sentinal value to indicate transaction was sanitized before epoch boundary
+                    Err(TransactionError::BlockhashNotFound)
```

Comment:
how about adding new variant at `TransactionError` like `ResanitizationNeeded`?

Reply:
Similarly, I also don't think this is necessary now since we are attempting re-sanitization on the worker side for epoch-crosses.